tests/integration/integration_test.cpp

```diff
@@ -211,11 +215,32 @@ int main() {
         sonata_check_and_flush(t);
         t += dt;
         // Change data every timestep
-        change_data(element_neurons);
-        change_data(soma_neurons);
-        change_data(single_neurons);
+        change_data(element_neurons, element_buffered_data);
+        change_data(soma_neurons, soma_buffered_data);
+        change_data(single_neurons, single_neurons_buffered_data);
     }
     sonata_flush(t);
+    sonata_clear();
+
+    // Write pre-buffered data (GPU usecase)
+    if (global_rank == 0) {
+        logger->info("GPU usecase: Writting prebuffered steps");
+    }
```

Comment:
just curious - what is this part? I assume nothing is formalised for gpu use case?

Reply:
I took the work of writing prebuffered steps from the draft PR I had regarding GPU usecase:
https://bbpteam.epfl.ch/project/issues/browse/REP-71
https://github.com/BlueBrain/libsonatareport/pull/18